docs/architecture.md

```diff
@@ -77,6 +77,31 @@ class BusinessLogicComponent extends Bloc {
 }
 ```
 
+## Bloc State Subscription
+
+> ​Bloc state subscription is a practice used to allow Bloc to react to state changes happened in another bloc.
+
+Inside the same Business Logic Layer, some Blocs may need to communicate with each other. in this case `state.listen` can be used to get notified when bloc state changes.
+
+```dart
+class FilteredBusinessLogicComponent extends Bloc {
+  final BusinessLogicComponent businessLogicComponent;
+  StreamSubscription todosSubscription;
+
+  FilteredBusinessLogicComponent({@required this.businessLogicComponent}) {
+    todosSubscription = businessLogicComponent.state.listen((state) {
+		// react to state change happened in another Bloc
```

Comment:
can we fix the indentation here so that it's consistent with the rest of the code?